lib/session.ts

```diff
@@ -396,6 +396,15 @@ export class Session extends Entity {
         const msg: string = `Unable to create the amqp receiver '${receiver.name}' on amqp ` +
           `session '${this.id}' due to operation timeout.`;
         log.error("[%s] %s", this.connection.id, msg);
+
+        const createReceiverOptions = options as CreateReceiverOptions;
+        if (createReceiverOptions?.session?.createReceiver) {
+          // being called on a session passed via the options so don't close the session
+          receiver.close({ closeSession: false }).then(() => { receiver.remove(); })
+        } else {
+          receiver.close({ closeSession: true }).then(() => { receiver.remove(); })
+        }
```

Comment:
[NIT]

Should this just be extracted into a receiverCleanup so that it can be reused?

Reply:
Good point! But could wait until it is going to be re-used.